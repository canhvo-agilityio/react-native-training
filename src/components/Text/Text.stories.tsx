import type { Meta, StoryObj } from '@storybook/react';

import Text from '.';

const meta: Meta<typeof Text> = {
  title: 'components/Text',
  component: Text,
};

export default meta;

type Story = StoryObj<typeof Text>;

export const Default: Story = {
  args: {
    value: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
};

export const Heading: Story = {
  args: {
    value: 'HEADING TEXT',
    variant: 'heading',
    size: '2xl',
  },
};

export const Title: Story = {
  args: {
    value: 'This is title',
    variant: 'title',
    size: 'xs',
  },
};

export const Description: Story = {
  args: {
    value: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    variant: 'description',
    size: 'xs',
  },
};
