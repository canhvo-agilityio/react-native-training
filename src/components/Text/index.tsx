import { ReactNode, memo } from 'react';
import {
  ColorValue,
  Text as RNText,
  TextProps as RNTextProps,
  StyleProp,
  TextStyle,
} from 'react-native';

// Themes
import { textSizes, textVariants } from '@/themes';

// Types
import { TextSizeTypes, TextVariantTypes } from '@/types';

export interface TextProps extends RNTextProps {
  value: string | ReactNode;
  variant?: TextVariantTypes;
  size?: TextSizeTypes;
  color?: ColorValue;
  style?: StyleProp<TextStyle>;
}

const Text = ({
  value,
  variant = 'default',
  size = 'sm',
  color,
  style,
  ...props
}: TextProps) => (
  <RNText
    style={[
      textVariants[variant],
      textSizes[size],
      style,
      { ...(color && { color }) },
    ]}
    {...props}>
    {value}
  </RNText>
);

export default memo(Text);
