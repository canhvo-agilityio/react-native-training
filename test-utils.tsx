import { PropsWithChildren, ReactElement } from 'react';
import { render } from '@testing-library/react-native';

const AllTheProviders = ({ children }: PropsWithChildren) => {
  return <>{children}</>;
};

const customRender = (
  ui: ReactElement,
  createNodeMock?: (element: React.ReactElement) => any,
) => render(ui, { wrapper: AllTheProviders, createNodeMock });

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };
