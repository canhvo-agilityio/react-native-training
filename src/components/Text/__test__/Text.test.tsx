import { render } from 'test-utils';

import Text from '..';

describe('Text', () => {
  const props = {
    value: 'Text content',
  };
  it('should render text component with default props', () => {
    const { toJSON } = render(<Text {...props} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render text component with variant heading', () => {
    const { toJSON } = render(<Text {...props} variant="heading" size="2xl" />);

    expect(toJSON()).toMatchSnapshot();
  });
});
