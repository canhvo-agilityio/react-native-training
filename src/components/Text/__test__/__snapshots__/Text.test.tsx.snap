// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render text component with default props 1`] = `
<Text
  style={
    [
      {
        "color": "#606060",
        "fontFamily": "NunitoSans-Regular",
      },
      {
        "fontSize": 14,
      },
      undefined,
      {},
    ]
  }
>
  Text content
</Text>
`;

exports[`Text should render text component with variant heading 1`] = `
<Text
  style={
    [
      {
        "color": "#303030",
        "fontFamily": "Merriweather-Bold",
      },
      {
        "fontSize": 30,
      },
      undefined,
      {},
    ]
  }
>
  Text content
</Text>
`;
